tests/integration/schema/reinitialize-associations-test.js

```diff
@@ -0,0 +1,38 @@
+import Mirage from 'ember-cli-mirage';
+import Model from 'ember-cli-mirage/orm/model';
+import Schema from 'ember-cli-mirage/orm/schema';
+import Db from 'ember-cli-mirage/db';
+import {module, test} from 'qunit';
+
+// Model classes are defined statically, just like in a typical app
+var User = Model.extend({
+  address: Mirage.hasMany()
+});
+var Address = Model.extend();
+
+module('Integration | Schema | reinitialize associations', {
+  beforeEach: function() {
+    this.db = new Db({
+      users: [],
+      addresses: []
+    });
+    this.schema = new Schema(this.db);
+
+    this.schema.registerModels({
+      address: Address,
+      user: User
+    });
```

Comment:
By the way in some tests I've started just instantiating `Server`, since the setup is a bit easier (ie test needs to know less about how to wire up schema/db). Wanted to share with you for future reference:

```js
this.server = new Server({
  environment: 'development',
  models: {
    author: Model.extend({
      posts: Mirage.hasMany(),
    }),
    post: Model
  }
});
this.server.timing = 0;
this.server.logging = false;
```

You can then work with schema via `this.server.schema`.